tst/EnergyPlus/unit/OASystemHWPreheatCoil.unit.cc

```diff
@@ -2020,6 +2022,11 @@ namespace EnergyPlus {
 		CpAir = PsyCpAirFnWTdb( Node( AirInletNodeNum ).HumRat, Node( AirInletNodeNum ).Temp );
 		EXPECT_NEAR( WaterCoil( 1 ).TotWaterHeatingCoilRate, WaterCoil( 1 ).InletAirMassFlowRate * CpAir * ( WaterCoil( 1 ).OutletAirTemp - WaterCoil( 1 ).InletAirTemp ), 1.0 ); 
 
+		// test that OA sys water coil bypasses normal controller calls before air loop simulation
+		EXPECT_EQ( "PREHEAT COIL CONTROLLER", HVACControllers::ControllerProps( 1 ).ControllerName );
+		EXPECT_TRUE( HVACControllers::ControllerProps( 1 ).BypassControllerCalc );
+		// test that water coil knows which controller controls the HW coil
+		EXPECT_EQ( WaterCoil( 1 ).ControllerIndex, 1 );
 	}
 
 }
```

Comment:
"TL;DR: because source code files can be imported and you want to ensure you don't accidentally glue code together."

Git is showing the warning, but really most programming languages have a warning mechanism for this.  The reason is that C++, like most, have an import capability.  In C++ you can `#include` other C++ files.  And this include isn't magic, it simply brings in the source of the other file directly at the spot of the include statement.  The include doesn't append a newline at the end of it's import, it relies on the user to do what's right in that situation.  Every situation I've ever seen is that you'd want a newline to avoid accidentally gluing code together with imports.  Not sure if that makes super sense; it's getting late.  So if you want more info, a Google of "why do source code files need trailing newlines?" would probably be better than my response tonight.